src/run/runner/helpers/apt.rs

```diff
@@ -0,0 +1,126 @@
+use super::run_with_sudo::run_with_sudo;
+use crate::prelude::*;
+use crate::run::check_system::SystemInfo;
+use std::path::Path;
+use std::process::Command;
+
+fn is_system_compatible(system_info: &SystemInfo) -> bool {
+    system_info.os == "ubuntu" || system_info.os == "debian"
+}
+
+pub fn install(system_info: &SystemInfo, packages: &[&str]) -> Result<()> {
+    if !is_system_compatible(system_info) {
+        bail!(
+            "Package installation is not supported on this system, please install necessary packages manually"
+        );
+    }
+
+    debug!("Installing packages: {packages:?}");
+
+    run_with_sudo(&["apt-get", "update"])?;
+    let mut install_cmd = vec![
+        "apt-get",
+        "install",
+        "-y",
+        "--allow-downgrades",
+        "--reinstall",
+    ];
+    install_cmd.extend_from_slice(packages);
+    run_with_sudo(&install_cmd)?;
+
+    info!("Packages installed successfully");
+    Ok(())
+}
+
+/// Restore cached tools from the cache directory to the root filesystem
+pub fn restore_from_cache(system_info: &SystemInfo, cache_dir: &Path) -> Result<()> {
+    if !is_system_compatible(system_info) {
+        warn!("Cache restore is not supported on this system, skipping");
+        return Ok(());
+    }
+
+    if !cache_dir.exists() {
+        debug!("Cache directory does not exist: {}", cache_dir.display());
+        return Ok(());
+    }
+
+    // Check if the directory has any contents
+    let has_contents = std::fs::read_dir(cache_dir)
+        .map(|mut entries| entries.next().is_some())
+        .unwrap_or(false);
+
+    if !has_contents {
+        debug!("Cache directory is empty: {}", cache_dir.display());
+        return Ok(());
+    }
+
+    info!(
+        "Restoring tools from cache directory: {}",
+        cache_dir.display()
+    );
+
+    // Use bash to properly handle glob expansion
+    let cache_dir_str = cache_dir
+        .to_str()
+        .ok_or_else(|| anyhow!("Invalid cache directory path"))?;
+
+    let copy_cmd = format!("cp -r {cache_dir_str}/* /");
+
+    let output = Command::new("sudo")
+        .args(["bash", "-c", &copy_cmd])
+        .output()
+        .context("Failed to run cache restore command")?;
+
```

Comment:
can you use run_with_sudo here? Avoiding manual sudos is important because atm it's not handled well if you need to type a password for example.
The custom command will make it possible to improve this behavior